### PR TITLE
Documentation search

### PR DESCRIPTION
**Pull request by attfarhan**
Index documentation to produce relevant results for natural language queries.

##### Reviewer tasks
- [ ] HACKS: reviewer approves hacks introduced by this change
- [ ] DOCS: reviewer approves the docstrings (or the justified omission thereof)
- [ ] UNIT-TEST: reviewer approves the unit tests (or the justified omission thereof)

### CODE REVIEW

**Comment by attfarhan** [comment]
Results from local instance with some queries:
http://cl.ly/1F3S1I3w0Q1y
http://cl.ly/40423C1J1k3B
http://cl.ly/1e1d1g1m1v2q

**Comment by beyang** [comment]
^ these look promising!

File: services/backend/internal/localstore/global_defs.go [hunk 1]
```diff
@@ -180,7 +180,12 @@ func (g *globalDefs) Search(ctx context.Context, op *store.GlobalDefSearchOp) (*
 		// The ranking critieron is the weighted sum of xref count,
 		// text similarity score, and whether the last term matches
 		// the name.
-		scoreSQL = `5.0*log(10 + ref_ct) + 100.0*ts_rank(to_tsvector('english', bow), to_tsquery('english', ` + arg(bowQuery) + `)) + 100.0*((LOWER(name)=LOWER(` + arg(lastTok) + `))::int) score`
+		if bowQuerySplit := strings.Split(bowQuery, " & "); len(bowQuerySplit) > 2 {
```
Comment (beyang) [inline services/backend/internal/localstore/global_defs.go]: Operating on the tsquery is hacky. Better to apply the transformation to the original (`op.TokQuery`). That way, the behavior is tied to what the user actually types into the search bar, not to something we've already transformed a bit for another purpose.

How many cases have you tested this out on? Is 2 the ideal split? What if I have a 3-component symbol that I'm looking for (e.g., gorilla/mux.Router)? Does search still work well in that situation? Can you feature flag this so we can play around with it? E.g., you can enable it only for admin users (like here: https://github.com/sourcegraph/sourcegraph/blob/cb259045902836e6de77508f1f420f933ff4d80f/app/web_modules/sourcegraph/tree/TreeSearch.js#L197)

**Comment by beyang** [comment]
@attfarhan I'll take this over this week so you can focus other stuff

**Comment by beyang** [comment]
fixes https://github.com/sourcegraph/sourcegraph-issues/issues/23

**Pull request closed without merging**